src/pages/SupplierInvoicePage.js

```diff
@@ -107,56 +208,234 @@ export class SupplierInvoicePage extends GenericPage {
           title: `${pageInfoStrings.confirm_date}:`,
           info: formatDate(transaction.confirmDate),
         },
+        {
+          title: `${pageInfoStrings.total_price}:`,
+          info: this.state.totalPrice,
+        },
       ],
       [
         {
+          title: `${pageInfoStrings.supplier}:`,
+          info: this.props.transaction.otherParty && this.props.transaction.otherParty.name,
+        },
+        {
           title: `${pageInfoStrings.their_ref}:`,
           info: transaction.theirRef,
+          onPress: this.openTheirRefEditor,
+          editableType: 'text',
         },
         {
           title: `${pageInfoStrings.comment}:`,
           info: transaction.comment,
+          onPress: this.openCommentEditor,
+          editableType: 'text',
         },
       ],
     ];
-    return <PageInfo columns={infoColumns} />;
+    return (
+      <PageInfo columns={infoColumns} isEditingDisabled={this.props.transaction.isFinalised} />
+    );
   }
 
-  renderCell(key, transactionItem) {
+  renderCell(key, transactionBatch) {
+    const isEditable = !this.props.transaction.isFinalised;
+    const type = isEditable ? 'editable' : 'text';
+    const editableCell = {
+      type: type,
```

Comment:
What's the point of making an external supplier invoice if you can say how much you received? Where else would the quantity value come for an external invoice?

In both cases the quantities in the unfinalised invoices needs to be editable, so that the receiving store can process it, make adjustments and finalise. This leaves a trail of any discrepancies that have occurred.

In terms of the other fields, being able to edit and correct and discrepancies is what is needed - it might be a mistake, it might be foul play, but without the ability for the receiver to make changes there isn't a discernible audit trail.

Quick non-foul example: Supplier claimed to send batch A, customer received batch B. Batch B has a recall because bad. Good thing receiving store corrected the received batch according to label/inspection before finalising supplier invoice, now they aren't dispensing a bad batch.